plugin/pkg/managedseed/validator/admission.go

```diff
@@ -0,0 +1,463 @@
+		}
+		// TODO If spec.DNS.Provider is not nil should we check if the user specified exactly the DNS provider we found above?
+		// I assume not, since the shoot may have multiple DNS providers
+
+		if spec.Ingress.Domain == "" {
+			spec.Ingress.Domain = ingressDomain
+		} else if spec.Ingress.Domain != ingressDomain {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ingress", "domain"), spec.Ingress.Domain, fmt.Sprintf("seed ingress domain must be equal to shoot DNS domain %s", ingressDomain)))
+		}
+	} else {
+		if spec.DNS.IngressDomain == nil || *spec.DNS.IngressDomain == "" {
+			spec.DNS.IngressDomain = &ingressDomain
+		} else if *spec.DNS.IngressDomain != ingressDomain {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("dns", "ingressDomain"), spec.DNS.IngressDomain, fmt.Sprintf("seed ingress domain must be equal to shoot DNS domain %s", ingressDomain)))
+		}
+	}
+
+	// Initialize and validate networks
+	if spec.Networks.Nodes == nil {
+		spec.Networks.Nodes = shoot.Spec.Networking.Nodes
+	} else if shoot.Spec.Networking.Nodes != nil && *spec.Networks.Nodes != *shoot.Spec.Networking.Nodes {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "nodes"), spec.Networks.Nodes, fmt.Sprintf("seed nodes CIDR must be equal to shoot nodes CIDR %s", *shoot.Spec.Networking.Nodes)))
+	}
+	if spec.Networks.Pods == "" && shoot.Spec.Networking.Pods != nil {
+		spec.Networks.Pods = *shoot.Spec.Networking.Pods
+	} else if shoot.Spec.Networking.Pods != nil && spec.Networks.Pods != *shoot.Spec.Networking.Pods {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "pods"), spec.Networks.Pods, fmt.Sprintf("seed pods CIDR must be equal to shoot pods CIDR %s", *shoot.Spec.Networking.Pods)))
+	}
+	if spec.Networks.Services == "" && shoot.Spec.Networking.Services != nil {
+		spec.Networks.Services = *shoot.Spec.Networking.Services
+	} else if shoot.Spec.Networking.Services != nil && spec.Networks.Services != *shoot.Spec.Networking.Services {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "services"), spec.Networks.Pods, fmt.Sprintf("seed services CIDR must be equal to shoot services CIDR %s", *shoot.Spec.Networking.Services)))
+	}
+
+	// Initialize and validate provider
+	if spec.Provider.Type == "" {
+		spec.Provider.Type = shoot.Spec.Provider.Type
+	} else if spec.Provider.Type != shoot.Spec.Provider.Type {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("provider", "type"), spec.Provider.Type, fmt.Sprintf("seed provider type must be equal to shoot provider type %s", shoot.Spec.Provider.Type)))
+	}
+	if spec.Provider.Region == "" {
+		spec.Provider.Region = shoot.Spec.Region
+	} else if spec.Provider.Region != shoot.Spec.Region {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("provider", "region"), spec.Provider.Region, fmt.Sprintf("seed provider region must be equal to shoot region %s", shoot.Spec.Region)))
+	}
+
+	return allErrs, nil
```

Comment:
I think we should also cross check that exactly one of the seed VPA and shoot VPA is enabled, but not both and not none of both.

Reply:
This is done by the controller. Unfortunately, we can't check it here because we don't have access to the seed. Back in the old PR, @rfranzke proposed some sophisticated mechanism based on new shoot conditions that would allow us to do it here, however I don't believe the outcome warrants the additional complexity.

Currently it works like this: the ManagedSeed would be accepted and stored, but the controller would recognize that both VPAs are enabled and report an error, the managed seed would not get reconciled until one of the VPAs is disabled. 

So I would propose to merge the admission plugin as it is and go back to the discussion during the review of the controller PR, if needed.

Reply:
Yeah, the problem with the VPA is the "migration phase". It's a bit special and cannot be tackled here already.

Reply:
Ok, sgtm. Thanks for clarifying.